### PR TITLE
pkg/system.getxattr: handle changed size case

### PR DESCRIPTION
**Pull request by kolyshkin**
lgetxattr(2) man page says:

> If size is specified as zero, these calls return the  current  size  of
> the  named extended attribute (and leave value unchanged).  This can be
> used to determine the size of the buffer that should be supplied  in  a
> subsequent  call.   (But, bear in mind that there is a possibility that
> the attribute value may change between the two calls,  so  that  it  is
> still necessary to check the return status from the second call.)

The current code does not handle the "But..." case, i.e. when the size
changes between the two calls, and the new size is larger.

Fix the above problem, and slightly simplify the code flow.

### CODE REVIEW

File: pkg/system/xattrs_linux.go [hunk 1]
```diff
@@ -10,24 +10,23 @@ func Lgetxattr(path string, attr string) ([]byte, error) {
 	dest := make([]byte, 128)
 	sz, errno := unix.Lgetxattr(path, attr, dest)
 
-	switch {
-	case errno == unix.ENODATA:
-		return nil, nil
-	case errno == unix.ERANGE:
-		// 128 byte array might just not be good enough. A dummy buffer is used
-		// to get the real size of the xattrs on disk
+	for errno == unix.ERANGE {
```
Comment (thaJeztah) [inline pkg/system/xattrs_linux.go]: I don't see one directly, but is there a potential situation where this ends up in an infinite loop? And/or does this need throttling?

File: pkg/system/xattrs_linux.go [hunk 2]
```diff
@@ -10,24 +10,23 @@ func Lgetxattr(path string, attr string) ([]byte, error) {
...
+		// Buffer too small, get the real size first
```
Comment (thaJeztah) [inline pkg/system/xattrs_linux.go]: Perhaps keep info about the dummy buffer, or would that be obvious?

```suggestion
		// Buffer too small, use a dummy buffer to get the real size of the xattrs on disk
```

File: pkg/system/xattrs_linux.go (hunk 1 above)
Comment (kolyshkin) [reply pkg/system/xattrs_linux.go]: There's a very slight chance that we'll ever get an `ERANGE` for the second time, and if we'll get it more than that it means the kernel is severely broken, and the world is coming to the end...

File: pkg/system/xattrs_linux.go (hunk 2 above)
Comment (kolyshkin) [reply pkg/system/xattrs_linux.go]: Makes sense; patch updated. The key here is we're using a buffer with size of 0.

File: pkg/system/xattrs_linux.go (hunk 1 above)
Comment (kolyshkin) [reply pkg/system/xattrs_linux.go]: Throttling is not needed here either, this is not supposed to fail with ERANGE repeatedly. 

Here's the flow:

1. we get the attributes using the default buffer size of 128 bytes.
2. if the actual size is larger (ERANGE), we
2.1 get the actual size
2.2 get the attributes using the buffer of the actual size

Now, in an _extremely_ rare scenario that the xattrs changes between 2.1 and 2.2, and the new ones are larger, we need to repeat 2 one more time.

Comment (thaJeztah) [reply pkg/system/xattrs_linux.go]: ❤️ Thanks for your in-depth answer! (better to ask, than to overlook 🤗). From your answer, I agree that the corner cases would be _very_ unlikely, so not something to care about 👍

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@kolyshkin was this fix addressing an actual issue that you encountered (and a backport needed), or mostly a theoretical situation?

**Comment by kolyshkin** [comment]
> @kolyshkin was this fix addressing an actual issue that you encountered (and a backport needed), or mostly a theoretical situation?

This is all theoretical (so no need to backport), although I guess there's a chance to actually reproduce this bug in real life.

**Pull request merged**